snp_motion_planning/src/plugins/tasks/constant_tcp_speed_time_parameterization_task.cpp

```diff
@@ -85,45 +86,59 @@ class ConstantTCPSpeedTimeParameterizationTask : public tesseract_planning::Task
 protected:
   friend struct tesseract_common::Serialization;
   friend class boost::serialization::access;
-  tesseract_planning::TaskComposerNodeInfo::UPtr runImpl(tesseract_planning::TaskComposerContext& context,
-                                                         OptionalTaskComposerExecutor /*executor*/) const override final
+  tesseract_planning::TaskComposerNodeInfo runImpl(tesseract_planning::TaskComposerContext& context,
+                                                   OptionalTaskComposerExecutor /*executor*/) const override final
   {
-    auto info = std::make_unique<tesseract_planning::TaskComposerNodeInfo>(*this);
-    info->return_value = 0;
+    tesseract_planning::TaskComposerNodeInfo info(*this);
+    info.return_value = 0;
+    info.status_code = 0;
 
-    // Get the problem
-    auto& problem = dynamic_cast<tesseract_planning::PlanningTaskComposerProblem&>(*context.problem);
+    if (context.isAborted())
+    {
+      info.status_message = "Aborted";
+      return info;
+    }
 
     // --------------------
     // Check that inputs are valid
     // --------------------
-    auto input_data_poly = context.data_storage->getData(input_keys_[0]);
-    if (input_data_poly.isNull() ||
-        input_data_poly.getType() != std::type_index(typeid(tesseract_planning::CompositeInstruction)))
+    auto env_poly = getData(*context.data_storage, INPUT_ENVIRONMENT_PORT);
+    if (env_poly.getType() != std::type_index(typeid(std::shared_ptr<const tesseract_environment::Environment>)))
     {
-      info->status_message = "Input results to constant TCP speed time parameterization must be a composite "
-                             "instruction";
-      CONSOLE_BRIDGE_logError("%s", info->status_message.c_str());
+      info.status_code = 0;
+      info.status_message = "Input data '" + input_keys_.get(INPUT_ENVIRONMENT_PORT) + "' is not correct type";
+      CONSOLE_BRIDGE_logError("%s", info.status_message.c_str());
+      info.return_value = 0;
```

Comment:
```suggestion
      info.color = "red";
      info.status_message = "Input data '" + input_keys_.get(INPUT_ENVIRONMENT_PORT) + "' is not correct type";
      CONSOLE_BRIDGE_logError("%s", info.status_message.c_str());
```

The status code and results values should already be set to 0, so no need to reset them here